analyzer/codechecker_analyzer/cmd/fixit.py

```diff
@@ -0,0 +1,299 @@
+    """
+
+    return {
+        'prog': 'CodeChecker fixit',
+        'formatter_class': arg.RawDescriptionDefaultHelpFormatter,
+        'description': """
+Some analyzers may suggest some automatic bugfixes. Most of the times these are
+style issues which can be fixed easily. This command handles the listing and
+application of these automatic fixes.
+
+Besides the provided filter options you can pipe the JSON format output of
+"CodeChecker cmd diff" command to apply automatic fixes only for new reports:
+CodeChecker cmd diff -b dir1 -n dir2 -o json --new | CodeChecker fixit dir2""",
+        'help': "Apply automatic fixes based on the suggestions of the "
+                "analyzers"
+    }
+
+
+def add_arguments_to_parser(parser):
+    """
+    Add the subcommand's arguments to the given argparse.ArgumentParser.
+    """
+
+    parser.add_argument('input',
+                        type=str,
+                        nargs='+',
+                        metavar='folder',
+                        help="The analysis result folder(s) containing "
+                             "analysis results and fixits which should be "
+                             "applied.")
+    parser.add_argument('-l', '--list',
+                        action='store_true',
+                        default=argparse.SUPPRESS,
+                        help="List the available automatic fixes.")
+    parser.add_argument('-i', '--interactive',
+                        action='store_true',
+                        default=False,
+                        help="Interactive selection of fixits to apply. Fixit "
+                             "items are enumerated one by one and you may "
+                             "choose which ones are to be applied.")
+    parser.add_argument('--checker-name',
+                        nargs='*',
+                        default=[],
+                        help='Filter results by checker names. The checker '
+                             'name can contain multiple * quantifiers which '
+                             'matches any number of characters (zero or '
+                             'more). So for example "*DeadStores" will '
+                             'match "deadcode.DeadStores".')
+    parser.add_argument('--file',
+                        metavar='FILE_PATH',
+                        nargs='*',
+                        default=[],
+                        help='Filter results by file path. The file path can '
+                             'contain multiple * quantifiers which matches '
+                             'any number of characters (zero or more). So if '
+                             'you have /a/x.cpp and /a/y.cpp then "/a/*.cpp" '
+                             'selects both.')
+
+    logger.add_verbose_arguments(parser)
+    parser.set_defaults(func=main)
+
+
+def get_location_by_offset(filename, offset):
+    """
+    This function returns the line and column number in the given file which
```

Comment:
What is the offset here? Number or bytes, lines, words?

Reply:
Characters, bytes, or something like that... I'm a bit worried about UTF-8 characters :smile: But I tried and it worked too.